src/Features/Core/Portable/Completion/Providers/AbstractSymbolCompletionProvider.cs

```diff
@@ -153,67 +102,44 @@ private bool ShouldIncludeInTargetTypedCompletionList(ISymbol symbol, ImmutableA
-            {
-                var symbols = originatingContextMap.Keys;
-                var symbolGroups = from symbol in symbols
-                                   let texts = GetDisplayAndSuffixAndInsertionText(symbol, originatingContextMap[symbol])
-                                   group symbol by texts into g
-                                   select g;
+            var symbolGroups = from symbol in symbols
+                               let texts = GetDisplayAndSuffixAndInsertionText(symbol, contextLookup(symbol))
+                               group symbol by texts into g
+                               select g;
 
-                var itemListBuilder = ImmutableArray.CreateBuilder<CompletionItem>();
-
-                foreach (var symbolGroup in symbolGroups)
-                {
-                    var item = this.CreateItem(
-                            symbolGroup.Key.displayText, symbolGroup.Key.suffix, symbolGroup.Key.insertionText, symbolGroup.ToList(),
-                            originatingContextMap[symbolGroup.First()], invalidProjectMap, totalProjects, preselect);
-
-                    item = AddTargetTypeMatchTagIfAppropriate(item, originatingContextMap, inferredTypes, symbolGroup);
-
-                    itemListBuilder.Add(item);
-                }
+            var itemListBuilder = ImmutableArray.CreateBuilder<CompletionItem>();
 
-                return itemListBuilder.ToImmutable();
-            }
-            else
+            foreach (var symbolGroup in symbolGroups)
             {
-                var symbols = originatingContextMap.Keys;
-                var q = from symbol in symbols
-                        let texts = GetDisplayAndSuffixAndInsertionText(symbol, originatingContextMap[symbol])
-                        group symbol by texts into g
-                        select this.CreateItem(
-                            g.Key.displayText, g.Key.suffix, g.Key.insertionText, g.ToList(),
-                            originatingContextMap[g.First()], invalidProjectMap, totalProjects, preselect);
-
-                return q.ToImmutableArray();
-            }
-        }
+                var arbitraryFirstContext = contextLookup(symbolGroup.First());
+                var item = this.CreateItem(
+                        symbolGroup.Key.displayText, symbolGroup.Key.suffix, symbolGroup.Key.insertionText, symbolGroup.ToList(),
+                        arbitraryFirstContext, invalidProjectMap, totalProjects, preselect);
 
-        private CompletionItem AddTargetTypeMatchTagIfAppropriate(
-            CompletionItem item,
-            Dictionary<ISymbol, SyntaxContext> originatingContextMap,
-            ImmutableArray<ITypeSymbol> inferredTypes,
-            IGrouping<(string displayText, string suffix, string insertionText), ISymbol> symbolGroup)
-        {
-            foreach (var symbol in symbolGroup)
-            {
-                foreach (var syntaxContext in originatingContextMap.Values)
```

Comment:
Deleting this code is also fixing a potentially big perf problem: to determine if a target type matched, we were accidentally binding each symbol up to _n_ times, where _n_ is the number of _total_ symbols in the list. This should have been only finding the syntaxContexts for the symbol by looking it up in the dictionary, but instead by matching all was binding far more than it needed to.